### PR TITLE
Special Pages: Malware warning support

### PR DESCRIPTION
**Pull request by mgurgel**
**Asana Task/Github Issue:** https://app.asana.com/0/0/1208033567421355/f

## Description

Adds Malware warning page to Special Error pages. 

`initialSetup` message changed to accommodate additional params. More info here: https://app.asana.com/0/1206594217596623/1208824527069247/f

## Testing Steps

1. View the previews: 
    1. [Phishing](https://deploy-preview-1268--content-scope-scripts.netlify.app/build/pages/special-error/?errorId=phishing)
    2. [Malware](https://deploy-preview-1268--content-scope-scripts.netlify.app/build/pages/special-error/?errorId=malware)
4. Copy, layout and functionality must match [Figma](https://www.figma.com/design/uP27mEGEaHCI7ZYXAs1815/Safe-Browsing-Warnings?node-id=1117-11982&m=dev)

## Checklist

<!—
  These questions are a friendly reminder to shipping code, if you're uncertain ask the AoR owners.
  It's also totally appropriate to not check some of these boxes, if they don't apply to your change.
—>
*Please tick all that apply:*

- [ ] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

### CODE REVIEW

File: special-pages/pages/special-error/app/hooks/ErrorStrings.jsx [hunk 1]
```diff
@@ -2,11 +2,44 @@ import { h } from 'preact';
...
+    href: phishingMalwareHelpPageURL,
+    target: '_blank',
+});
+
+/** @type {(url: string) => AnchorTagParams} */
+const reportSiteAnchorTagParams = (url) => {
+    const sanitizedURL = sanitizeAndEncodeURL(url);
+
+    return {
+        href: `${reportSiteAsSafeFormURL}${sanitizedURL}`,
```
[the remark this answers is not in the document]
Comment (mgurgel) [reply special-pages/pages/special-error/app/hooks/ErrorStrings.jsx:40]: Of course 🤦

File: special-pages/pages/special-error/app/hooks/ErrorStrings.jsx [hunk 2]
```diff
@@ -2,11 +2,44 @@ import { h } from 'preact';
...
+ * @typedef {object} AnchorTagParams
+ * @property {string} href
+ * @property {string} target
+ */
+
+/** @type {() => AnchorTagParams} */
+const helpPageAnchorTagParams = () => ({
+    href: phishingMalwareHelpPageURL,
+    target: '_blank',
+});
```
[the remark this answers is not in the document]
Comment (mgurgel) [reply special-pages/pages/special-error/app/hooks/ErrorStrings.jsx:33]: It doesn’t, but I was aiming at some consistency between that and `reportSiteAnchorTagParams`, but you’re right. No real benefit.

File: special-pages/pages/special-error/app/hooks/ErrorStrings.jsx [hunk 3]
```diff
@@ -2,11 +2,44 @@ import { h } from 'preact';
...
+    if (!urlString) return '';
+
+    try {
+        const url = new URL(urlString);
+
+        return encodeURIComponent(`${url.origin}${url.pathname}`);
+    } catch (error) {
+        return '';
+    }
+};
```
[the remark this answers is not in the document]
Comment (mgurgel) [reply special-pages/pages/special-error/app/hooks/ErrorStrings.jsx]: Tbh I didn’t question it much, which I should have. But in hindsight, it made sense for robustness. If in the future one of the platforms introduces a bug where it starts passing query params into the error page — if we're not being strict with what we pass down, the query params would make their way all the way down to the form.

File: special-pages/tests/page-objects/special-error.js [hunk 4]
```diff
@@ -254,6 +259,24 @@ export class SpecialErrorPage {
...
+        ).toBeVisible();
+        await this.showsAdvancedInfo();
+        await expect(
+            page.getByText(
+                'If you believe this website is safe, you can report an error. You can still visit the website at your own risk.',
+                { exact: true },
+            ),
+        ).toBeVisible();
+    }
+
```
Comment (shakyShane) [inline special-pages/tests/page-objects/special-error.js:279]: for future PRs, let's look into ariaSnapshots in playwright, i think it'll make these (excellent) tests easier to write

**Review by shakyShane — APPROVED** [review]
Nice work!

**Review by shakyShane — DISMISSED** [review]

**Review by shakyShane — APPROVED** [review]

**Pull request merged**